pkg/cdi/cdi.go

```diff
@@ -80,12 +84,20 @@ func (c *impl) CreateCDISpecForPool(resourcePrefix string, rPool types.ResourceP
 		cdiSpec.Devices = append(cdiSpec.Devices, device)
 	}
 
-	err = cdi.GetRegistry().SpecDB().WriteSpec(&cdiSpec, cdiSpecPrefix+resourcePrefix)
+	name, err := cdi.GenerateNameForSpec(&cdiSpec)
+	if err != nil {
+		glog.Errorf("CreateCDISpecForPool(): can not generate transient name: %v", err)
+		return err
+	}
+
+	err = cdi.GetRegistry().SpecDB().WriteSpec(&cdiSpec, cdiSpecPrefix+name+"-"+rPool.GetResourceName())
```

Comment:
cant we use : `strings.Join("-", cdiSpecPrefix, resourcePrefix, rPool.GetResourceName)` as name ?

Reply:
I don't really have a preference. My thought was that it's best to use cdi provided helper, as we use it already to write the spec to file.

Reply:
ack, your call.